eventuate-core/src/main/scala/com/rbmhtechnology/eventuate/ReplicationEndpoint.scala

```diff
@@ -363,6 +359,96 @@ class ReplicationEndpoint(
+    sourceFilters: Map[String, ReplicationFilter],
+    targetSourceCombinator: (ReplicationFilter, ReplicationFilter) => ReplicationFilter) extends EndpointFilter {
+
+    override def filterFor(targetLogId: String, sourceLogName: String): ReplicationFilter = {
+      (targetFilters.get(targetLogId), sourceFilters.get(sourceLogName)) match {
+        case (None, Some(sourceFilter))               => sourceFilter
+        case (Some(targetFilter), None)               => targetFilter
+        case (None, None)                             => NoFilter
+        case (Some(targetFilter), Some(sourceFilter)) => targetSourceCombinator(targetFilter, sourceFilter)
+      }
+    }
+  }
+
+  /**
+   * Creates an [[EndpointFilter]] that computes a [[ReplicationFilter]] for a replication read request
+   * from a source log to a target log by and-combining target and source filters when given in the provided [[Map]]s.
+   * If only source or target filter is given that is returned. If no filter is given [[com.rbmhtechnology.eventuate.ReplicationFilter.NoFilter]] is returned.
+   *
+   * This [[EndpointFilter]] can be useful when an eventsourced service that is not replicated collaborates
+   * with different external service. In that case `sourceFilters` can ensure that no ''internal'' event is
+   * replicated to other services and `targetFilters` can add target specific filter criteria.
+   *
+   * @param targetFilters maps target log ids to the [[ReplicationFilter]] that shall be applied when replicating to the target log
+   * @param sourceFilters maps source log names to the [[ReplicationFilter]] that shall be applied when replicating from the source log
+   */
+  def targetAndSourceFilters(targetFilters: Map[String, ReplicationFilter], sourceFilters: Map[String, ReplicationFilter]): EndpointFilter =
+    new CombiningEndpointFilter(targetFilters, sourceFilters, _ and _)
+
+  /**
+   * Creates an [[EndpointFilter]] that computes a [[ReplicationFilter]] for a replication read request
+   * from a source log to a target log by returning the target filter when given in `targetFilters`. If only
+   * a source filter is given in `sourceFilters` that is returned otherwise [[com.rbmhtechnology.eventuate.ReplicationFilter.NoFilter]] is returned.
+   *
+   * This [[EndpointFilter]] can be useful when an eventsourced service that can be replicated collaborates
+   * with different internal services and has a fixed set of events for external communication.
+   * In that case `sourceFilters` can ensure that only external events are replicated to external services
+   * while in case of internal communication (with internal service or replication partners) these filters
+   * can be overwritten.
+   *
+   * @param targetFilters maps target log ids to the [[ReplicationFilter]] that shall be applied when replicating to the target log
+   * @param sourceFilters maps source log names to the [[ReplicationFilter]] that shall be applied when replicating from the source log
```

Comment:
Distinguishing *replicating to the target log* from *replicating from the source log* is a bit misleading here. In both case there is a replication from the source log to the target log.